### PR TITLE
Add date_start and date_end for openmx parser

### LINKED ISSUE #1: OpenMX: parse timing information
https://nomad-lab.eu/prod/v1/staging/gui/analyze/metainfo/nomad.datamodel.datamodel.EntryArchive/run/time_run/date_start start is printed at the beginning of out.

https://nomad-lab.eu/prod/v1/staging/gui/analyze/metainfo/nomad.datamodel.datamodel.EntryArchive/run/time_run/date_end 
is date_start + Elapsed.Time. which is printed at the end.

### PR DESCRIPTION
**Pull request by jfrodlova**
Fixes ondracka/electronic-parsers#1

### CODE REVIEW

File: electronicparsers/openmx/parser.py [hunk 1]
```diff
@@ -164,6 +165,20 @@ def convert_eigenvalues(string):
 
 mainfile_parser = TextParser(
     quantities=[
+        Quantity(
+            'date_start',
+            r'\s+([A-Z]{1}[a-z]{2}\s[A-Z]{1}[a-z]{2}\s\d{1,2}\s\d{2}:\d{2}:\d{2}\s\d{4})',
+            #r'\s+(Sat Apr 17 18:14:52 2021)',
```
Comment (ondracka) [inline electronicparsers/openmx/parser.py]: Leftover debug stuff.

File: electronicparsers/openmx/parser.py [hunk 2]
```diff
@@ -716,6 +731,18 @@ def parse(self, mainfile: str, archive: EntryArchive, logger):
         sec_run.program = Program(
             name='OpenMX', version=str(mainfile_parser.get('program_version'))
         )
+        date_start = mainfile_parser.get('date_start')
+
+        elapsed_time = mainfile_parser.get('elapsed_time')
```
Comment (ondracka) [inline electronicparsers/openmx/parser.py]: We should have some checks that the regex actually returns the reasonable (non None) values (the reason might be corrupted output, or some output differences that the parser is not prepared for). Specifically for the date_start this is not and issue, its printed quite early and not much can go wrong, but the elapsed_time is printed at the end, so there is a reasonable assumption that it is not there if something goes wrong during the calculation. So we should check if `mainfile_parser.get('elapsed_time')` is None and in that case don't save `sec_run.time_run.date_end` and skip any calculations that would be using the values.

**Review by ondracka — CHANGES_REQUESTED** [review]
This is looking OK, but needs tests, look for example in https://github.com/ondracka/electronic-parsers/commit/402b0e11ec5e593e39c1b151ab8cbc55c98c397a how it could look like. We won't be adding any new testcases here, just extend 2 or 3 of the current cases that are already run to check for the new fields.

BTW you can enable the automatic tests in your fork as well, the test setup is defined in     electronic-parsers/.github/workflows/python-actions.yaml you somehow need to enable this in the github actions in your fork (I managed to do it somehow for my fork in the past, but I don't remember anymore). Maybe @mfrodl knows how this works. Or if you can't figure out, you can always push to my fork and that should start the CI automatically on the branch.

File: electronicparsers/openmx/parser.py (hunk 2 above)
Comment (jfrodlova) [reply electronicparsers/openmx/parser.py]: OK, sure. I  added :
```
if elapsed_time is not None:
    date_end = date_start_timestamp + elapsed_time
```
Is it what you meant?

Comment (ondracka) [reply electronicparsers/openmx/parser.py]: Almost  :wink:  `sec_run.time_run.date_end = date_end` should be also guarded in the if.

**Comment by ondracka** [comment]
This looks fine, just few nitpicks

- Its OK to add the tests in separate commits, but it should contain just the tests (adding the commented line regex in one commit and removing it in other one is not a good practice, although NOMAD upstream developers don't particularly care about having a clean history, so the regex removal should be amended to the first commit and the added if as well). Or you can just squash both, but this can be a good practice in git rebasing :sweat_smile: 

I assume pytest passes locally? If you can't get the github actions working to have it run automatically on push, just push the branch to my fork (you should have access). Besides pytest, there are some other other tests (formatting with ruff for example, but that should not be an issue if you used IDE, that usually takes care of auto-formating).

```
    date_start_datetime = datetime(2021, 4, 17, 18, 14, 52)
    date_start_timestamp = date_start_datetime.timestamp()
    assert run.time_run.date_start.magnitude == date_start_timestamp
```
The temporary variables feel quite unnecessary, I would just write it in one long (properly line-breaked if needed) expression `assert run.time_run.date_start.magnitude == datetime(2021, 4, 17, 18, 14, 52).timestamp()` but this is up to you.

File: electronicparsers/openmx/parser.py [hunk 3]
```diff
@@ -732,17 +731,16 @@ def parse(self, mainfile: str, archive: EntryArchive, logger):
...
 
-        sec_run.time_run = TimeRun()
-        sec_run.time_run.date_end = date_end
-        sec_run.time_run.date_start = date_start_timestamp
+        elapsed_time = mainfile_parser.get('elapsed_time')
+        if elapsed_time is not None:
+            date_end = date_start_timestamp + elapsed_time
+            sec_run.time_run = TimeRun()
+            sec_run.time_run.date_end = date_end
+            sec_run.time_run.date_start = date_start_timestamp
```
Comment (ondracka) [inline electronicparsers/openmx/parser.py]: Actually, `sec_run.time_run = TimeRun()` and `sec_run.time_run.date_start = date_start_timestamp` should not be in the if block, they make sense even if we fail to parse the elapsed time.

**Comment by jfrodlova** [comment]
> I assume pytest passes locally? If you can't get the github actions working to have it run automatically on push, just push the branch to my fork (you should have access). Besides pytest, there are some other other tests (formatting with ruff for example, but that should not be an issue if you used IDE, that usually takes care of auto-formating).
> 

Yes, it passes locally. I didn't manage to enable automatic tests on GitHub yet. I can push the branch to your fork later today.

**Comment by ondracka** [comment]
> > I assume pytest passes locally? If you can't get the github actions working to have it run automatically on push, just push the branch to my fork (you should have access). Besides pytest, there are some other other tests (formatting with ruff for example, but that should not be an issue if you used IDE, that usually takes care of auto-formating).
> 
> Yes, it passes locally. I didn't manage to enable automatic tests on GitHub yet. I can push the branch to your fork later today.

Please do. Once the tests are green, I think we are ready to create PR at the upstream repo https://github.com/nomad-coe/electronic-parsers and wait for upstream feedback. In the meantime you want to revisit https://github.com/ondracka/nomad/issues/6 or another parser task?

EDIT: by "we are ready to create PR" I mean "you will create". :smile:

**Comment by ondracka** [comment]
This was merged upstream already.

**Pull request closed without merging**